packages/database/src/tvl/value/repository.ts

```diff
@@ -46,6 +48,40 @@ export class ValueRepository extends BaseRepository {
     return rows.map(toRecord)
   }
 
+  async getLatestValues(projectIds?: ProjectId[]) {
+    if (projectIds?.length === 0) return []
+    const latestValuesQuery = (cb: QueryCreator<DB>) => {
+      let query = cb.selectFrom('public.values').select([
+        ...selectValue,
+        this.db.fn
+          .agg('ROW_NUMBER')
+          .over((over) =>
+            over
+              .partitionBy(['project_id', 'data_source'])
+              .orderBy('timestamp', 'desc'),
+          )
+          .as('combination_number'),
+      ])
+      if (projectIds) {
+        query = query.where(
+          'project_id',
+          'in',
+          projectIds.map((id) => id.toString()),
+        )
+      }
+
+      return query
+    }
+
+    const rows = await this.db
+      .with('latest_values', latestValuesQuery)
+      .selectFrom('latest_values')
+      .select(selectValue)
+      .where('combination_number', '=', 1)
+      .execute()
+    return rows.map(toRecord)
+  }
+
```

Comment:
We have almost 1:1 duplicate down below for da-beat - Let's either unify these or split this apart removing the optional param (`projectIds?: ProjectId[]`)

Reply:
yup, i wanted to unify that